go-api/aggregate/controller.go

```diff
@@ -0,0 +1,173 @@
+package aggregate
+
+import (
+	"encoding/json"
+	"go-api/utils"
+	"time"
+
+	"github.com/go-playground/validator/v10"
+	"github.com/gofiber/fiber/v2"
+	"github.com/gofiber/fiber/v2/log"
+)
+
+type AggregateRedisValueModel struct {
+	Timestamp *utils.CustomDateTime `db:"timestamp" json:"timestamp" validate:"required"`
+	Value     *utils.CustomInt64    `db:"value" json:"value" validate:"required"`
+}
+
+type WrappedInsertModel struct {
+	Data AggregateInsertModel `json:"data"`
+}
+
+type AggregateInsertModel struct {
+	Timestamp    *utils.CustomDateTime `db:"timestamp" json:"timestamp" validate:"required"`
+	Value        *utils.CustomInt64    `db:"value" json:"value" validate:"required"`
+	AggregatorId *utils.CustomInt64    `db:"aggregator_id" json:"aggregatorId" validate:"required"`
+}
+
+type AggregateModel struct {
+	AggregateId  *utils.CustomInt64    `db:"aggregate_id" json:"id"`
+	Timestamp    *utils.CustomDateTime `db:"timestamp" json:"timestamp" validate:"required"`
+	Value        *utils.CustomInt64    `db:"value" json:"value" validate:"required"`
+	AggregatorId *utils.CustomInt64    `db:"aggregator_id" json:"aggregatorId" validate:"required"`
+}
+
+type AggregateIdModel struct {
+	AggregateId *utils.CustomInt64 `db:"aggregate_id" json:"id"`
+}
+
+func insert(c *fiber.Ctx) error {
+	_payload := new(WrappedInsertModel)
+	if err := c.BodyParser(_payload); err != nil {
+		log.Panicf("failed to parse body:" + err.Error())
+		return c.Status(fiber.StatusBadRequest).SendString(err.Error())
```

Comment:
@nick-bisonai `log.Panicf` is a shortcut for `Printf` + `panic`. The return statement will not be reached.

https://pkg.go.dev/log#Panicf